### ISSUE TITLE
The new user registration form is not accessible to users using a screen reader

### ISSUE THREAD

**Issue by Andrea-Guevara** [DSpace/dspace-angular#3271]
## Describe the bug
The new user registration form is not accessible to users using a screen reader. 

## To reproduce
1. Using a screen reader, go to the login area.
2. Click on “New user? Click here to register”.
3. Navigate through the “New user registration” page and check that the screen reader doesn't clearly tell the user what to do, or even can't identify the error messages.

## Expected behavior
The functionalities of the page could be more accessible and intuitive for users who use a screen reader. For example: a better description of what the user needs to do, error messages need to be identified by the screen reader and even if disabled the “register” button needs to be accessible.

**Comment by tdonohue** [comment]
Closing as this was fixed by #3272
